crates/astria-sequencer-client/src/client.rs

```diff
@@ -17,95 +18,41 @@ use eyre::{
 };
 use tendermint::block::Height;
 use tendermint_rpc::{
-    endpoint::{
-        block::Response as BlockResponse,
-        broadcast::{
-            tx_commit,
-            tx_sync,
-        },
-        validators,
+    endpoint::broadcast::{
+        tx_commit,
+        tx_sync,
     },
     Client as _,
-    HttpClient,
 };
 
 /// Default Tendermint base URL.
 pub const DEFAULT_TENDERMINT_BASE_URL: &str = "http://localhost:26657";
 
 /// Tendermint HTTP client which is used to interact with the Sequencer node.
-pub struct Client {
-    client: HttpClient,
+#[async_trait]
+pub trait Client {
```

Comment:
Let's call this `ClientExt` to make clear that it is an extension trait. It should also be defined as `pub trait ClientExt: tendermint_rpc::Client`.

Look at `TryFutureExt` as an example of how that's done: https://docs.rs/futures-util/0.3.28/src/futures_util/future/try_future/mod.rs.html#137